src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java

```diff
@@ -1285,6 +1285,7 @@ private ElementKind getAllowedKind(ElementKind kind) {
                 case CONSTRUCTOR: case ENUM_CONSTANT: case EXCEPTION_PARAMETER:
                 case FIELD: case INSTANCE_INIT: case LOCAL_VARIABLE: case PARAMETER:
                 case RESOURCE_VARIABLE: case STATIC_INIT: case TYPE_PARAMETER:
+                case RECORD_COMPONENT:
```

Comment:
I'm not saying this is wrong, but I'd like to understand why it is necessary.

Reply:
HtmlDocletWriter.getPreviewNotes analyzes classes and their members, to see if some are using aspects that are under preview. When looking at the members, it uses utils.isIncluded on the member, and that eventually gets here. And if the member is a RECORD_COMPONENT, it would fail here. But we can avoid asking for RECORD_COMPONENTS, if needed.

Reply:
ok